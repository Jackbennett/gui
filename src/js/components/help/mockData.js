export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
