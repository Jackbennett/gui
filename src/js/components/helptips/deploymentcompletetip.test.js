import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import DeploymentCompleteTip from './deploymentcompletetip';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('DeploymentCompleteTip Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      devices: {
        byId: {},
        byStatus: { accepted: { deviceIds: [] } },
        filters: []
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeploymentCompleteTip />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
