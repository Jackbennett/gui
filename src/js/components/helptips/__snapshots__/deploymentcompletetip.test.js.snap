// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentCompleteTip Component renders correctly 1`] = `
<div>
  <p>
    Fantastic! You completed your first deployment!
  </p>
  <p>
    Your deployment is finished and your device is now running the updated software!
  </p>
  <div
    className="flexbox centered"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Go to http://localhost:85
      </span>
    </button>
  </div>
  <p>
    and you should see the demo web application actually being run on the device.
  </p>
  <p>
    NOTE: if you have local network restrictions, you may need to check them if you have difficulty loading the page.
  </p>
  <a
    onClick={[Function]}
  >
    Visit the web app running your device
  </a>
</div>
`;
